[New Term] C++ Math Functions: `exp2()` (#1089)
Codecademy/docs: content/cpp/concepts/math-functions/terms/exp2/exp2.md

---
Title: 'exp2()'
Description: 'Returns the base-2 exponential function, i.e. 2 raised to the given argument.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`exp2()`** is a built-in function in C++ STL that computes the `base-2` exponential function of a given number, i.e. `2` raised to the given argument.

## Syntax

```pseudo
exp2(x)
```

 - It takes a single mandatory argument (can be positive, negative or 0).
 - The `exp2()` function returns the value in the range of `[0, ∞]`.

>  - If no errors occur, the base-2 exponential of `n` is returned.
>  - If a range error due to overflow occurs, `HUGE_VAL`, `HUGE_VALF`, or `HUGE_VALL` is returned.
>  - If a range error occurs due to underflow, the correct result (after rounding) is returned.
>  - If the argument is ±0, 1 is returned.
>  - If the argument is -∞, +0 is returned.
>  - If the argument is +∞, +∞ is returned.
>  - If the argument is NaN, NaN is returned.
>  - The return type for the `exp2()` function is `double`, `float`, or `long double`.

Reviewer: ```suggestion
 - If a range error due to overflow occurs, `HUGE_VAL`, `HUGE_VALF`, or `HUGE_VALL` is returned.
 - If a range error occurs due to underflow, the correct result (after rounding) is returned.
 - If the argument is ±0, 1 is returned.
 - If the argument is -∞, +0 is returned.
 - If the argument is +∞, +∞ is returned.
 - If the argument is `NaN`, `NaN` is returned.
 - The return type for the `exp2()` function is `double`, `float`, or `long double`.
```

---
Title: 'exp2()'

> Description: 'Returns the base-2 exponential function, i.e. 2 raised to the given argument'

Reviewer 2: ```suggestion
Description: 'Returns the base-2 exponential function, i.e. 2 raised to the given argument.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`exp2()`** is a builtin function in C++ STL that computes the `base-2` exponential function of a given number, i.e. `2` raised to the given argument.

Reviewer 2: ```suggestion
The **`exp2()`** is a built-in function in C++ STL that computes the `base-2` exponential function of a given number, i.e. `2` raised to the given argument.
```

## Syntax

```pseudo
exp2(x)
```

It takes a single mandatory argument (can be positive, negative or 0).
The `exp2()` function returns the value in the range of `[0, ∞]`.

> If no errors occur, the base-2 exponential of `n` is returned.

Reviewer 2: These lines, in addition to the "Error Handling" lines should all be in a single bulleted list.

Author: Please check if the list is as expected.

If a range error due to overflow occurs, `HUGE_VAL`, `HUGE_VALF`, or `HUGE_VALL` is returned.
If a range error occurs due to underflow, the correct result (after rounding) is returned.
<details>
	<summary>Error Handling</summary>

	If the argument is ±0, 1 is returned.
	If the argument is -∞, +0 is returned.
	If the argument is +∞, +∞ is returned.
	If the argument is NaN, NaN is returned.
</details>

> The return type for the `exp2()` function is `double`, `float`, `long double`.

Reviewer 2: ```suggestion
The return type for the `exp2()` function is `double`, `float`, or `long double`.
```
